test/test_function_schema.py

```diff
@@ -112,6 +112,53 @@ def test_string_optional_parameter_default_value(self):
         schema_b = parse_schema(str(schema_a))
         self.assertEquals(schema_a, schema_b)
 
+    def test_forward_compatible_arguments_without_out(self):
+        old_schema = parse_schema('any(Tensor self, int a, int b=1) -> Tensor')
+        new_schema = parse_schema('any(Tensor self, int a) -> Tensor')
+        is_fc, _ = new_schema.check_forward_compatible_with(old_schema)
+        self.assertTrue(is_fc)
+        new_schema = parse_schema('any(Tensor self, int a, int b=1, int c=1) -> Tensor')
+        is_fc, _ = new_schema.check_forward_compatible_with(old_schema)
+        self.assertTrue(is_fc)
+        new_schema = parse_schema('any(Tensor self, int a, int b=1, int[2] c=1) -> Tensor')
+        is_fc, reason = new_schema.check_forward_compatible_with(old_schema)
+        self.assertFalse(is_fc)
+        self.assertEqual(reason, "Function schema is not forward compatible since the new argument"
```

Comment:
Is there a good test for type narrowing/widening?

Reply:
I added one test where we widen the types of start and end of aten::slice which actually caused production issue before :)